src/entities/longtime_attack.rs

```diff
@@ -0,0 +1,43 @@
+use crate::entities::data::{AttackableFrom, Damage, EntityType};
+use crate::entities::event::EntityDamageEvent;
+use bevy::app::{App, Plugin, Update};
+use bevy::prelude::{Component, Entity, EventWriter, Query};
+use bevy_xpbd_2d::prelude::CollidingEntities;
+
+pub struct LongTimeAttackPlugin;
+
+impl Plugin for LongTimeAttackPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_systems(Update, damage_entities);
+    }
+}
+
+#[derive(Component)]
+pub struct LongTimeAttack {
+    pub damaged_entities: Vec<Entity>,
+}
+
+fn damage_entities(
+    mut entity_attacks: Query<(&CollidingEntities, &mut LongTimeAttack, &Damage)>,
+    attackable_from: Query<&AttackableFrom>,
```

Comment:
this should properly include `With<Health>` since everything that can be attacked has health

```suggestion
    attackable_from: Query<&AttackableFrom, With<Health>>,
```